src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -174,59 +288,60 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
             {
-                auto actionAndArgs = ActionAndArgs::FromJson(actionJson, warnings);
-
-                if (actionAndArgs)
-                {
-                    result->_setAction(*actionAndArgs);
-                }
-                else
-                {
-                    // Something like
-                    //      { name: "foo", action: "unbound" }
-                    // will _remove_ the "foo" command, by returning null here.
-                    return nullptr;
-                }
-
-                // If an iterable command doesn't have a name set, we'll still just
-                // try and generate a fake name for the command give the string we
-                // currently have. It'll probably generate something like "New tab,
-                // profile: ${profile.name}". This string will only be temporarily
-                // used internally, so there's no problem.
-                result->_setName(_nameFromJsonOrAction(json, actionAndArgs));
+                result->_ActionAndArgs = *ActionAndArgs::FromJson(actionJson, warnings);
             }
             else
             {
                 // { name: "foo", action: null } will land in this case, which
                 // should also be used for unbinding.
-                return nullptr;
+
+                // create an "invalid" ActionAndArgs
+                result->_ActionAndArgs = make<implementation::ActionAndArgs>();
+            }
+
+            // GH#4239 - If the user provided more than one key
+            // chord to a "keys" array, warn the user here.
+            // TODO: GH#1334 - remove this check.
+            const auto keysJson{ json[JsonKey(KeysKey)] };
+            if (keysJson.isArray() && keysJson.size() > 1)
+            {
+                warnings.push_back(SettingsLoadWarnings::TooManyKeysForChord);
+            }
+            else
+            {
+                Control::KeyChord keys{ nullptr };
+                if (JsonUtils::GetValueForKey(json, KeysKey, keys))
+                {
+                    result->RegisterKey(keys);
+                }
             }
         }
-        else
-        {
-            result->_setName(_nameFromJson(json));
-        }
+
+        // If an iterable command doesn't have a name set, we'll still just
+        // try and generate a fake name for the command give the string we
+        // currently have. It'll probably generate something like "New tab,
+        // profile: ${profile.name}". This string will only be temporarily
+        // used internally, so there's no problem.
+        result->_name = _nameFromJson(json);
 
         // Stash the original json value in this object. If the command is
         // iterable, we'll need to re-parse it later, once we know what all the
         // values we can iterate on are.
         result->_originalJson = json;
 
-        if (result->_Name.empty())
-        {
-            return nullptr;

```

Comment:
this seems an important check

Reply:
This used to be important because `Command` was only used for the command palette. So if a command didn't have a name, it wasn't relevant to the command palette.

That's no longer the case. An Action Map will _always_ care about what you (a `Command`) have to say. So if you have no name, good for you. But we still care about your keys and ActionAndArgs.